File: app/src/main/java/io/github/takusan23/kaisendon/Zyanken/ZyankenMenu.java
package io.github.takusan23.kaisendon.Zyanken;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.LinearLayout;

import io.github.takusan23.kaisendon.R;

public class ZyankenMenu extends AppCompatActivity {

    //Button
    Button host_Button;
    Button client_Button;
    //LienarLayout
    LinearLayout zyanken_LinearLayout;

    //枠をあける
    Button start_host_button;
    EditText start_host_EditText;
    //参加する
    Button start_client_button;
    EditText start_client_EditText;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_zyanken_menu);

        Button host_Button = findViewById(R.id.zyanken_host);
        Button client_Button = findViewById(R.id.zyanken_join);
        LinearLayout zyanken_LinearLayout = findViewById(R.id.zyanken_menu_linearLayout);

        //枠をあける
        host_Button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                //テキストボックスとボタンを動的に生成
                start_host_button = new Button(ZyankenMenu.this);
                start_host_button.setText("枠をあける");
                start_host_EditText = new EditText(ZyankenMenu.this);
                //Intent
                Intent intent = new Intent(ZyankenMenu.this, ZyankenSetup.class);
                intent.putExtra("mode", "host");
                startActivity(intent);
            }
        });

        //参加用
        client_Button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                //テキストボックスとボタンを動的に生成
                start_client_button = new Button(ZyankenMenu.this);
                start_client_button.setText("枠をあける");
                start_client_EditText = new EditText(ZyankenMenu.this);
                //Intent
                Intent intent = new Intent(ZyankenMenu.this, Zyanken.class);
                intent.putExtra("mode", "client");
                startActivity(intent);
            }
        });

    }
}

File: app/src/main/java/io/github/takusan23/Kaisendon/Zyanken/ZyankenMenu.kt
package io.github.takusan23.Kaisendon.Zyanken

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.LinearLayout
import androidx.appcompat.app.AppCompatActivity
import io.github.takusan23.Kaisendon.R

class ZyankenMenu : AppCompatActivity() {

    //Button
    internal var host_Button: Button? = null
    internal var client_Button: Button? = null
    //LienarLayout
    internal var zyanken_LinearLayout: LinearLayout? = null

    //枠をあける
    internal lateinit var start_host_button: Button
    internal lateinit var start_host_EditText: EditText
    //参加する
    internal lateinit var start_client_button: Button
    internal lateinit var start_client_EditText: EditText

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_zyanken_menu)

        val host_Button = findViewById<Button>(R.id.zyanken_host)
        val client_Button = findViewById<Button>(R.id.zyanken_join)
        val zyanken_LinearLayout = findViewById<LinearLayout>(R.id.zyanken_menu_linearLayout)

        supportActionBar!!.title = "じゃんけんメニュー"

        //枠をあける
        host_Button.setOnClickListener {
            //テキストボックスとボタンを動的に生成
            start_host_button = Button(this@ZyankenMenu)
            start_host_button.text = "枠をあける"
            start_host_EditText = EditText(this@ZyankenMenu)
            //Intent
            val intent = Intent(this@ZyankenMenu, ZyankenSetup::class.java)
            intent.putExtra("mode", "host")
            startActivity(intent)
        }

        //参加用
        client_Button.setOnClickListener {
            //テキストボックスとボタンを動的に生成
            start_client_button = Button(this@ZyankenMenu)
            start_client_button.text = "枠をあける"
            start_client_EditText = EditText(this@ZyankenMenu)
            //Intent
            val intent = Intent(this@ZyankenMenu, Zyanken::class.java)
            intent.putExtra("mode", "client")
            startActivity(intent)
        }
    }


}
